### PR TITLE
feat: added shp file to metadata conversion

### LINKED ISSUE #4: Feat: read shape files and merge them into metadata model
all relevant shape files for given layer should be parsed and merged into metadata model

### PR DESCRIPTION
**Pull request by cl-shaharshavit-rnd**
| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix         | ✖                                                                        |
| New feature     | ✔                                                                       |
| Breaking change | ✖                                                                        |
| Deprecations    | ✖                                                                        |
| Documentation   | ✖                                                                        |
| Tests added     | ✖                                                                        |
| Chore            | ✖                                                                       |

Closes #4 

Further  information:

### CODE REVIEW

File: src/layerCreator/models/trigger.ts [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+            const productGeoJson = await this.shpParser.parse(productShp,productDbf);
+            const metaDataGeoJson = await this.shpParser.parse(metadataShp,metadataDbf);
+            this.parseToMetadata(productGeoJson,metaDataGeoJson,files);
+            //TODO: trigger overseer
+            //TODO: update history status to triggered
+        }
+    }
+
+    private async fileExists(path: string): Promise<boolean>{
+        return access(path, fsConstants.F_OK)
```
Comment (asafMasa) [inline src/layerCreator/models/trigger.ts:69]: have you considered using https://www.npmjs.com/package/fs-extra?

File: src/layerCreator/models/trigger.ts [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+        //TODO: if history don't exists create it
+        //TODO: if history is not pending exit
+        //check if all shp files exists
+        const filesShp = `${directory}/Files.shp`;
+        const filesDbf = `${directory}/Files.dbf`;
+        const productShp = `${directory}/Product.shp`;
+        const productDbf = `${directory}/Product.dbf`;
+        const metadataShp = `${directory}/ShapeMetadata.shp`;
+        const metadataDbf = `${directory}/ShapeMetadata.dbf`;
+        if(await this.fileExists(filesShp) && await this.fileExists(filesDbf) &&
```
Comment (asafMasa) [inline src/layerCreator/models/trigger.ts:44]: why not https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all?

File: src/layerCreator/models/trigger.ts [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+@injectable()
+export class Trigger{
+public constructor(private readonly shpParser: ShpParser){}
+    public async trigger(directory: string){
+        //TODO: get history
+        //TODO: if history don't exists create it
+        //TODO: if history is not pending exit
+        //check if all shp files exists
+        const filesShp = `${directory}/Files.shp`;
```
Comment (asafMasa) [inline src/layerCreator/models/trigger.ts:38]: Refactor this code to a higher level methods such as: 
validateMetadataFilesExists() => check for metadata files
const metaInJson = parseMetaData() => parse the metadata to JSON format
validateImageFilesExists() => check for images files

File: src/layerCreator/models/trigger.ts [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+      private parseFilesShpJson(filesJson:GeoJSON): string[]{
+        const features = (filesJson as FeatureCollection).features;
+        const files = features.map((feature) =>{
+          // eslint-disable-next-line @typescript-eslint/naming-convention
+          const file = feature.properties as {"File Name": string,Format: string};
+          return `${file['File Name']}.${file.Format}`;
+        })
+        return files;
+      }
+}
```
Comment (asafMasa) [inline src/layerCreator/models/trigger.ts:102]: end of file

File: src/layerCreator/models/trigger.ts [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+        //check if all shp files exists
+        const filesShp = `${directory}/Files.shp`;
+        const filesDbf = `${directory}/Files.dbf`;
+        const productShp = `${directory}/Product.shp`;
+        const productDbf = `${directory}/Product.dbf`;
+        const metadataShp = `${directory}/ShapeMetadata.shp`;
+        const metadataDbf = `${directory}/ShapeMetadata.dbf`;
+        if(await this.fileExists(filesShp) && await this.fileExists(filesDbf) &&
+        await this.fileExists(productShp) && await this.fileExists(productDbf) &&
+        await this.fileExists(metadataShp) && await this.fileExists(metadataDbf)) {
```
Comment (asafMasa) [inline src/layerCreator/models/trigger.ts:46]: Raise proper error for missing files

**Review by asafMasa — CHANGES_REQUESTED** [review]
👍

File: src/layerCreator/models/trigger.ts (hunk 1 above)
Comment (cl-shaharshavit-rnd) [reply src/layerCreator/models/trigger.ts:69]: is it worth adding additional dependency for that?

File: src/layerCreator/models/trigger.ts (hunk 2 above)
Comment (cl-shaharshavit-rnd) [reply src/layerCreator/models/trigger.ts:44]: most of the times it is called files will be missing and I don't want to wait to check all of them if I don't have to

File: src/layerCreator/models/trigger.ts (hunk 5 above)
Comment (cl-shaharshavit-rnd) [reply src/layerCreator/models/trigger.ts:46]: missing file is not an error.
it will usually meant the copy of files is still in progress

Comment (asafMasa) [reply src/layerCreator/models/trigger.ts]: Split manual trigger and watch error handling - if manually triggered throw an error for missing file

File: src/layerCreator/models/trigger.ts [hunk 6]
```diff
@@ -1,102 +1,125 @@
...
+      //read file list
+      const filesGeoJson = await this.shpParser.parse(filesShp, filesDbf);
+      const files = this.parseFilesShpJson(filesGeoJson);
+      if (!(await this.validateLayerFilesExists(directory, files))) {
+        return;
+      }
+      // parse all shp files and convert to model
+      const productGeoJson = await this.shpParser.parse(productShp, productDbf);
+      const metaDataGeoJson = await this.shpParser.parse(metadataShp, metadataDbf);
+      this.parseToMetadata(productGeoJson, metaDataGeoJson, files);
```
Comment (asafMasa) [inline src/layerCreator/models/trigger.ts:54]: const metadata = ...

**Review by asafMasa — CHANGES_REQUESTED** [review]

**Review by AnguIar — APPROVED** [review]

**Pull request merged**